codegen/src/lib.rs

```diff
@@ -0,0 +1,82 @@
+#![recursion_limit = "128"]
+extern crate proc_macro;
+extern crate proc_macro2;
+
+use proc_macro::TokenStream;
+
+use quote::quote;
+use syn;
+use syn::Error;
+use syn::ItemFn;
+
+/// Procedural attribute macro. This is meant to be applied to a binary's async
+/// `main()`, transforming into a function that returns a type acceptable for
+/// `main()`. In other words, this will not compile with libtock-rs:
+///     async fn main() {}
+/// and this will:
+///     #[async_main]
```

Comment:
Now that the macro is `libtock::main`, this should be updated. I would like this PR merged sooner rather than later, so I'll fix this in my panic!()/LED PR.